### PR TITLE
[eclipse/xtext#1897] converted example tests to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1897] converted example tests to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractCodeMiningTest.java [hunk 1]
```diff
@@ -71,7 +72,7 @@ protected XtextEditor openEditor(IFile dslFile) {
 
 	protected void codeMiningsArePresent(XtextEditor editor, String expected) {
 		String actual = insertCodeMinings(editor);
-		assertEquals(expected, actual);
+		assertEquals(Strings.toUnixLineSeparator(expected), Strings.toUnixLineSeparator(actual));
```
Comment (cdietrich) [inline org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractCodeMiningTest.java:75]: i think this code change is ok

Comment (szarnekow) [reply org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractCodeMiningTest.java:75]: So do I.

File: org.eclipse.xtext.xtext.ui.examples/projects/arithmetics/org.eclipse.xtext.example.arithmetics.ui.tests/src/org/eclipse/xtext/example/arithmetics/ui/tests/autoedit/AutoEditTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	@Override
+	protected String getFileExtension() {
+		return extensionProvider.getPrimaryFileExtension();
+	}
+
+	private IProject createPluginProject(String name) {
+		projectFactory.setBreeToUse(JREContainerProvider.PREFERRED_BREE);
+		projectFactory.setProjectName(name);
+		projectFactory.addFolders(Collections.<String>singletonList("src"));
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/arithmetics/org.eclipse.xtext.example.arithmetics.ui.tests/src/org/eclipse/xtext/example/arithmetics/ui/tests/autoedit/AutoEditTest.java:70]: Nit: Unnecessary type argument.

File: org.eclipse.xtext.xtext.ui.examples/projects/arithmetics/org.eclipse.xtext.example.arithmetics.ui.tests/src/org/eclipse/xtext/example/arithmetics/ui/tests/autoedit/AutoEditTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+				"module MyModule\n" +
+				"\n" +
+				"1 + 2;\n" +
+				"// = 3\n" +
+				"|\n";
+		assertState(editor, expectation);
+	}
+
+	@Inject
+	private FileExtensionProvider extensionProvider;
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/arithmetics/org.eclipse.xtext.example.arithmetics.ui.tests/src/org/eclipse/xtext/example/arithmetics/ui/tests/autoedit/AutoEditTest.java:52]: If this wasn't a test I'd remark that the fields should be at the top of the class.

**Review by szarnekow — APPROVED** [review]

**Pull request merged**